pkg/nto/performanceprofile_test.go

```diff
@@ -0,0 +1,606 @@
+			expectedError:          nil,
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          fmt.Errorf("PerformanceProfile's name is empty"),
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:          nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		_, err := testCase.testPerformanceProfile.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testPerformanceProfile.Object)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestPerformanceProfileUpdate(t *testing.T) {
+	testCases := []struct {
+		testPerformanceProfile *Builder
+		topologyPolicy         string
+		forceFlag              bool
+		expectedError          string
+	}{
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			topologyPolicy:         defaultNumaTopology,
+			forceFlag:              true,
+			expectedError:          "",
+		},
+		{
+			testPerformanceProfile: buildValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			topologyPolicy:         defaultNumaTopology,
+			forceFlag:              false,
+			expectedError:          "",
+		},
+		{
+			testPerformanceProfile: buildInValidPerformanceProfileBuilder(buildPerformanceProfileWithDummyObject()),
+			expectedError:          "PerformanceProfile's name is empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Nil(t, nil, testCase.testPerformanceProfile.Definition.Spec.NUMA)
+		assert.Nil(t, nil, testCase.testPerformanceProfile.Object)
+		testCase.testPerformanceProfile.WithNumaTopology(testCase.topologyPolicy)
+		_, err := testCase.testPerformanceProfile.Update(testCase.forceFlag)
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, testCase.topologyPolicy,
+				*testCase.testPerformanceProfile.Definition.Spec.NUMA.TopologyPolicy)
+		}
+	}
+}
+
+func TestPerformanceProfileWithHugePages(t *testing.T) {
+	testCases := []struct {
+		testHugePageSize  string
+		testPages         []v2.HugePage
+		expectedErrorText string
+	}{
+		{
+			testHugePageSize:  defaultHugepageSize,
+			testPages:         defaultHugepages,
+			expectedErrorText: "",
+		},
+		{
+			testHugePageSize:  "1G",
+			testPages:         defaultHugepages,
+			expectedErrorText: "",
+		},
+		{
+			testHugePageSize:  "2G",
+			testPages:         defaultHugepages,
+			expectedErrorText: "'hugePageSize' argument is not in allowed list: [2M 1G]",
+		},
+		{
+			testHugePageSize:  "",
+			testPages:         defaultHugepages,
+			expectedErrorText: "'hugePageSize' argument cannot be empty",
+		},
+		{
+			testHugePageSize:  defaultHugepageSize,
+			testPages:         []v2.HugePage{},
```

Comment:
I'd consider adding 2 more tests:
* explicitly asking for 2 different NUMA nodes
```
        defaultHugePagesNodeOne = int32(0)
        defaultHugePagesNodeTwo = int32(1)
	defaultHugepages              = []v2.HugePage{
 		{
 			Size:  "2M",
 			Count: 32768,
 			Node:  &defaultHugePagesNodeOne,
 		},
                {
 			Size:  "2M",
 			Count: 32768,
 			Node:  &defaultHugePagesNodeTwo,
 		},
 	}
```
* not asking for a NUMA node
```
	defaultHugepages              = []v2.HugePage{
 		{
 			Size:  "2M",
 			Count: 32768,
 		},
        }
```